src/codeu/chat/util/Tokenizer.java

```diff
@@ -0,0 +1,73 @@
+package codeu.chat.util;
+
+import java.io.IOException;
+
+public final class Tokenizer {
+	private StringBuilder token;
+	private String source;
+	private int at;
+	
+	public Tokenizer(String source) {
+	  this.source = source;
+	  at = 0;
+	  token = new StringBuilder();
+    }
+
+	//TODO: add javadoc
+	/**
+	 * 
+	 * @return
+	 * @throws IOException
+	 */
+	public String next() throws IOException {
+	  // Skip all leading whitespace
+      while (remaining() > 0 && Character.isWhitespace(peek())) {
+        read();  // ignore the result because we already know that it is a whitespace character
+      }
+      if (remaining() <= 0) {
+        return null;
+      } else if (peek() == '"') {
```

Comment:
peek() returns a char, so this line isn't comparing strings. I could cast the char to a string and then use String.equals, but that feels weird. Let me know what makes the most sense to do :)